ingest/template/new_schema_template.py

```diff
@@ -106,21 +108,84 @@ def get_dictionary_representation(self):
+        :return: A string representing a fully qualified field name in the next latest metadata schema version if a
+                 migration to that schema version exists. If a migration does not exist, returns the original fully
+                 qualified key.
+        """
+
+        if not self._validate_fully_qualified_key_exists(fully_qualified_key):
+            raise Exception(f"ERROR: Fully qualified key {fully_qualified_key} does not exist in any of the schemas.")
+
+        if fully_qualified_key in self.migrations:
+            migration_info = self.migrations[fully_qualified_key]
+            return migration_info.replaced_by if migration_info.replaced_by in migration_info else fully_qualified_key
+
+        return fully_qualified_key
+
+    def lookup_absolute_latest_key_migration(self, fully_qualified_key):
+        """
+        Given a key, returns the fully qualified key in the maximal latest schema if it has been replaced.
+
+        :param key: A string representing a fully qualified field name in an older metadata schema version.
+        :return: A string representing a fully qualified field name in the latest metadata schema
+                 version if a migration chain to that schema version exists. If a migration does not exist, returns the
+                 original fully qualified key.
+        """
+
+        if not self._validate_fully_qualified_key_exists(fully_qualified_key):
+            raise Exception(f"ERROR: Fully qualified key {fully_qualified_key} does not exist in any of the schemas.")
+
+        original_key = fully_qualified_key
+        next_latest_migrated_key = self.lookup_next_latest_key_migration(fully_qualified_key)
+
+        while original_key != next_latest_migrated_key:
+            original_key = next_latest_migrated_key
+            next_latest_migrated_key = self.lookup_next_latest_key_migration(original_key)
+
+        return next_latest_migrated_key
+
+    def generate_yaml_representation_of_spreadsheets(self, tabs_only):
+        """
+        Generate a YAML object representing the schema_template object and if `tabs_only` is true, then only generate
+        the YAML based on self.tabs. This YAML object can be used to generate arbitrary spreadsheets based on the
+        metadata schemas.
+
+        :param tabs_only: When true, only uses the tabs to generate the YAML instead of including all the "metadata"
+                          about the metadata schemas encapsulated in this SchemaTemplate object.
+        :return: A YAML object representing the metadata schemas encapsulated in this SchemaTemplate object.
+        """
+
+        if tabs_only:
+            json_dump = json.dumps({"tabs": self.tabs}, indent=4)
+        else:
+            json_dump = json.dumps(self.get_dictionary_representation(), indent=4)
+
+        yaml.default_flow_style = False
+        return yaml.dump(yaml.load(json_dump, Loader=yaml.FullLoader))
```

Comment:
I think you could just store the dict and do 1 yaml.dump(dict, indent=4) in the end.

Reply:
Good catch! Fixed!